### DISCUSSION TITLE
Feature request for Spectrum Next (banked memory viewer)

### DISCUSSION THREAD

**Discussion by infromthecold** [maziac/DeZog#114] [General]
Hi
     I useful feature would be to add a new command to the memory viewer -mvb to display the memory from a bank, even when the breakpoint stops the code in another bank like -mvb $c000 200 BANK2 as this would be useful when looking for bank overwrites and other debugging banked code. 

Cheers

**Comment by maziac** [comment]
Hi, I see the usefulness of such a command but properly  the proposed command+arguments has a lot of drawbacks.
I.e. the bank could be paged in to different address offsets and it could also be paged in at different offsets at the same time.
Furthermore, maybe the bank is not paged in at all.
So it is probably only possible to show the memory starting at index 0 for each bank.
E.g. 
~~~
-mvbank <banknr> <index> <size>
~~~

(I would rather use 'bank' instead of just 'b' because I use the suffix 'b' often for a byte.)

~~~
-mvbank 2 $0100 20
~~~

would show 20 bytes of  memory of the bank 2 starting at index $100 inside the bank.

It would also not make sense to use labels. E.g.

~~~
-mvbank 2 main_label 20
~~~

would certainly not make sense.


But there is also a much bigger problem with the implementation:
The DZRP protocol that I use to communicate (with the ZXNext, CSpect and the internal simulator) does not contain such a command at the moment.
I can write to a bank, but I cannot read from a specific bank, unfortunately.
I.e. I would need to specify and implement it.
And that would mean the change is not only DeZog but also the dezogif (the zxnext program) and the cspect plugin.

Of course, all this is doable but I think I would wait for more changes in the protocol before doing so.

**Comment by maziac** [comment]
Will be in next release 3.8:
- Memory Views (debug commands "-mv", "-mvw" and "-mvd") allow additional parameter "bank=" to specify the bank to display.
- Debug commands "-md", "-mdelta", "-ml", "-ms", "-msetb" and "-msetw" allow additional parameter to specify a bank to read from or write to.
